Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java

```diff
@@ -85,6 +105,19 @@ public String showProject(Model model, @PathVariable("ProjectUnixName") String u
         if (principal != null) {
             user = principal.getName();
         }
+        // Fetch blog entries for a project identified by project unix name
+        String projectId = project.getProjectId();
+        Integer count = 1;
+        List<IProjectBlogEntry> projectBlogEntryList = projectBlogEntryManager.getProjectBlogEntryList(projectId,
+                count);
+        IProjectBlogEntry recentprojectblog = projectBlogEntryList.get(0);
+        String parseddescription = recentprojectblog.getDescription();
+        org.jsoup.nodes.Document dom = Jsoup.parse(parseddescription);
+        String text = dom.text();
+        parseddescription = firstWords(text, 40);              
+        recentprojectblog.setDescription(parseddescription);
```

Comment:
all of this code can then go away and you should be able to simply call your snipped method in the jsp page.